src/REPLMode.jl

```diff
@@ -58,9 +19,53 @@ end
 ###########
 # Options #
 ###########
+struct OptionSpec
+    name::String
+    short_name::String
+    is_meta::Bool
+    is_switch::Bool
+end
+
 @enum(OptionKind, OPT_ENV, OPT_PROJECT, OPT_MANIFEST, OPT_MAJOR, OPT_MINOR,
                   OPT_PATCH, OPT_FIXED, OPT_COVERAGE, OPT_NAME)
 
+const OptionDeclaration = Tuple(Union{String,Pair{String}}, Symbol, Symbol)
+declare_options = [
+    ("env", :meta, :arg)
+    ,(["project", "p"], :cmd, :switch)
+    ,(["manifest", "m"], :cmd, :switch)
+    ,("major", :cmd, :switch)
+    ,("minor", :cmd, :switch)
+    ,("patch", :cmd, :switch)
+    ,("fixed", :cmd, :switch)
+    ,("coverage", :cmd, :switch)
+    ,("name", :cmd, :switch)
```

Comment:
Leading commas like this are anti-idiomatic in Julia; Julia allows a comma after the last element, so just end every line with a comma. Note that vectors can also have no commas at all and just use newlines, but you have to be consistent.

Reply:
Thanks. I'll go with the trailing commas. (Also, I didn't know about the no commas for vectors. Pretty cool!)

Reply:
It's technically a different syntax: with commas it's a vector construction syntax, without commas its a vertical concatenation syntax, but in this case the result is the same. If the elements are themselves vectors or matrices, the result can be different, e.g.:
```jl
julia> [[1, 2],
        [3, 4, 5],
        [6, 7]]
3-element Array{Array{Int64,1},1}:
 [1, 2]
 [3, 4, 5]
 [6, 7]

julia> [[1, 2]
        [3, 4, 5]
        [6, 7]]
7-element Array{Int64,1}:
 1
 2
 3
 4
 5
 6
 7
```

Reply:
Ah, I see the difference now: no commas will end up calling `vcat`. Thanks for explaining